backend/pkg/kubeconfig/kubeconfig.go

```diff
@@ -281,33 +309,40 @@ func LoadContextsFromBase64String(kubeConfig string, source int) ([]Context, []e
 		return nil, []error{fmt.Errorf("error decoding base64 kubeconfig: %v", err)}
 	}
 
-	return loadContextsFromData(kubeConfigByte, source)
+	var skipProxySetup bool
+
+	if source == KubeConfig {
+		skipProxySetup = false
+	} else {
+		skipProxySetup = true
+	}
+
+	return loadContextsFromData(kubeConfigByte, source, skipProxySetup)
 }
 
-// loadContextsFromData is a helper function that loads contexts from a byte slice.
-// It parses the byte slice as a YAML file and validates the parsed data.
-// It then creates contexts from the parsed data and returns them.
-// It returns any errors that occurred during parsing or validation.
-func loadContextsFromData(data []byte, source int) ([]Context, []error) {
+// loadContextsFromData loads contexts from a kubeconfig data.
+// It unmarshals the kubeconfig data, extracts the contexts, and processes each context.
+// It returns the processed contexts and any errors that occurred during the process.
+func loadContextsFromData(data []byte, source int, skipProxySetup bool) ([]Context, []error) {
 	var contexts []Context //nolint:prealloc
 
 	var errs []error
 
-	// Parse and validate the config.
-	rawConfig, _, err := parseAndValidateConfig(data)
+	// Unmarshal the kubeconfig data
+	kubeconfig, err := UnmarshalKubeconfig(data)
 	if err != nil {
-		// Log the error but continue processing
-		errs = append(errs, fmt.Errorf("error parsing config: %v", err))
```

Comment:
We don't need to format the error anymore?

Reply:
It is now handled inside `UnmarshalKubeconfig` function